File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/likesdata/gateway/LikesDataGateway.java
package nl.acidcats.tumblrlikes.data_impl.likesdata.gateway;

import java.util.List;

import nl.acidcats.tumblrlikes.data_impl.likesdata.models.TumblrLikeVO;
import rx.Observable;

/**
 * Created by stephan on 28/03/2017.
 */

public interface LikesDataGateway {
    Observable<List<TumblrLikeVO>> getLikes(String blogName, int count, long beforeTime);
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/data_impl/likesdata/gateway/LikesDataGateway.kt
package nl.acidcats.tumblrlikes.data_impl.likesdata.gateway

import nl.acidcats.tumblrlikes.data_impl.likesdata.models.TumblrLikeVO
import rx.Observable

/**
 * Created by stephan on 28/03/2017.
 */

interface LikesDataGateway {
    fun getLikesPage(blogName: String, count: Int = 20, afterTime: Long): Observable<List<TumblrLikeVO>>

    val isLoadComplete: Boolean

    val lastLikeTimeSec : Long
}
